### ISSUE TITLE
Subtle bug in Remix guide

### ISSUE THREAD

**Issue by AlemTuzlak** [supabase/supabase#30900]
# Improve documentation

## Link 

 https://supabase.com/docs/guides/auth/social-login/auth-github?queryGroups=environment&environment=server&queryGroups=framework&framework=remix

## Describe the problem

This code is wrong:
```ts
const { data, error } = await supabase.auth.signInWithOAuth({
  provider,
  options: {
    redirectTo: 'http://example.com/auth/callback',
  },
})

if (data.url) {
  redirect(data.url) // use the redirect API for your server framework
}
```
It causes a bug where the cookie is not set and the auth does not work

## Describe the improvement

```ts
const headers = new Headers();

 const supabase = createServerClient(process.env.SUPABASE_URL!, process.env.SUPABASE_API_KEY!, { 
	cookies: {
		getAll() {
			return parseCookieHeader(request.headers.get("Cookie") ?? "")
		},
		setAll(cookiesToSet) {
			for (const { name, value, options } of cookiesToSet) {
				headers.append("Set-Cookie", serializeCookieHeader(name, value, options))
			}
		},
	},
})
const { data, error } = await supabase.auth.signInWithOAuth({
  provider,
  options: {
    redirectTo: 'http://example.com/auth/callback',
  },
})

if (data.url) { 
  redirect(data.url, { headers }) // use the redirect API for your server framework
}
```

## Additional context

The previous docs cause a subtle and hard to catch bug that cost my employee 2 days of work, and me 2 hours to figure out. It doesn't append the cookie headers to the redirect and doesn't set the cookies in the browser making the subsequent request to the callback url NOT have the verifier code or whatever it's called and making supabase throw an error

**Comment by charislam** [comment]
Thanks for opening the issue @AlemTuzlak!

We have an admonition about using the server-based client a little further above, but based on your feedback, we should make that more prominent. 👍🏼 

<img width="876" alt="image" src="https://github.com/user-attachments/assets/c555afa0-6bed-4273-8875-98a76b308b8d">

**Comment by AlemTuzlak** [comment]
@charislam the issue is not really about using the right client or not, the issue is that you don't pass the headers into the `redirect` which never sets them in the browser. This causes the auth to fail once you come back as you don't have the proper cookies set

**Comment by charislam** [comment]
@AlemTuzlak yep I get that! My point is that the link above _does_ direct users to set the right headers (it links to exactly the `createServerClient` code that you added on top). But we need to make it more prominent

(and maybe "using the right client" is not the clearest terminology 🤔)

**Comment by lurgi** [comment]
@AlemTuzlak Thank you so much. I can't even tell you how many hours I spent struggling with this. I thought I had to manually generate and inject the code_verifier myself. I wish the docs had explained this more clearly.

**Comment by AlemTuzlak** [comment]
@lurgi, you're welcome. It was a huge headscratcher for me as well, I was mad at my intern for not following the docs properly, and then I realised it was the docs 😅

**Comment by silentworks** [comment]
I personally think the comment was enough as you would then go and investigate how your framework works with redirecting a user and setting a cookie. But maybe it's just me, I've created a PR to add the `header` part to the `redirect`. I do think a user should have a basic understanding of their framework in order to get further with it. I didn't include where `header` comes from as this is a guide with snippets and not a tutorial.

**Comment by silentworks** [comment]
Since Remix is kinda unwinding now and React Router 7 is the upgrade path. Maybe we should update these guides to React Router 7 instead.

**Comment by tankit03** [comment]
Hello,

As someone who has used Supabase for hackathons, I know how critical clear auth guides are, and I see how this subtle issue could be frustrating. I've reviewed the problem and understand that the code example for signInWithOAuth in the Remix guide needs to be updated to pass the { headers } to the redirect function, this is what I see currently once I get assigned to it I would love to look into it deeply !

This would be a great first issue for me to tackle. Can you please assign it to me?

**Comment by abdulbaqui17** [comment]
Opened PR #43942 to fix this.

Summary of the docs update:
- Updated the shared server OAuth PKCE snippet to preserve Set-Cookie headers from @supabase/ssr.
- Updated the redirect example to pass headers via redirect(data.url, { headers }) for frameworks like Remix.

PR: https://github.com/supabase/supabase/pull/43942

**Comment by LeonxLJX** [comment]
I'd like to take this one.

Reading the issue, this looks like a documentation-clarity gap rather than a code defect: the current page doesn't give readers enough to self-diagnose, which is why it surfaces here as an issue instead of being answered by the docs.

Plan: locate the relevant section, confirm what the code actually does today (so the docs match behavior, not intent), tighten the wording, and scan neighboring pages for the same inconsistency — otherwise the fix just relocates the confusion.

One thing to confirm before writing: minimal correction scoped to this page, or also cross-link from related guides if the same gap shows up there?

(claiming via @LeonxLJX)

**Comment by cekuu35** [comment]
Hey AlemTuzlak,

I noticed this issue has been open for 1 years and I think I can help fix it.

**What I can do:**
- Fix the core issue with a clean, tested implementation
- Include tests to prevent regressions
- Handle edge cases and documentation

**Pricing:** $150 USD (crypto accepted: USDT, BTC, ETH)

**Delivery:** Within 6 hours of payment confirmation

**Includes:**
- Root cause analysis
- Bug fix implementation
- Unit and integration tests
- Code review comments
- Documentation updates
- Follow-up support (48h)

If you're interested, I can start right away. Just reply here and we can work out the details.

Thanks!

**Comment by aantti** [comment]
Hoping we're finally addressing it via PR #50722, apologies it took so long. Also, big kudos to everyone who have attempted to contribute a fix, I've left some notes in the PR.
